qiskit/circuit/quantumcircuit.py

```diff
@@ -6877,6 +6881,46 @@ def qubit_stop_time(self, *qubits: Union[Qubit, int]) -> float:
 
         return 0  # If there are no instructions over bits
 
+    def estimate_duration(self, target, unit: str = "s") -> int | float:
+        """Estimate the duration of a scheduled circuit
+
+        This method computes the estimate of the circuit duration by finding
+        the longest duration path in the circuit based on the durations
+        provided by a given target.
```

Comment:
We should include some description in this of the limitations of the function, beyond just the summary in the "raises" documentation - something like "this only works for simple circuits with no control flow or other classical feed-forwards operations".

Reply:
Added in: https://github.com/Qiskit/qiskit/pull/13783/commits/af6c739743060c3275fe418c84c12dab61830577